### PR TITLE
fix: ECS7 GLTF Component

### PR DESCRIPTION
**Pull request by pbosio**
*renamed `GLTFShape` for `GltfContainer`
*fixed collider creation for gltf
*apply renderers and colliders internal components
*created tests accordingly

### CODE REVIEW

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerHandler.cs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, PBGltfContainer model)
+        {
+            if (string.IsNullOrEmpty(model.Src) || prevLoadedGltf == model.Src)
+                return;
+
+            prevLoadedGltf = model.Src;
+
+            CleanUp(scene, entity);
+
+            gltfLoader.OnSuccessEvent += rendereable =>
```
Comment (Kinerius) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerHandler.cs:69]: Can we extract this into a Method?

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerHandler.cs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                    if (!meshFilters[i].transform.parent.name.ToLower().Contains("_collider"))
+                        continue;
+
+                    MeshCollider collider = meshFilters[i].gameObject.AddComponent<MeshCollider>();
+                    collider.sharedMesh = meshFilters[i].sharedMesh;
+                    meshFilters[i].gameObject.layer = PhysicsLayers.characterOnlyLayer;
+                    physicColliders.Add(collider);
+
+                    Object.Destroy(meshFilters[i].GetComponent<Renderer>());
+                }
```
Comment (Kinerius) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerHandler.cs:91]: Same with this, a CreatePhysicsColliders() would be nice and self explanatory

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerHandler.cs [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                        continue;
+
+                    GameObject colliderGo = new GameObject(POINTER_COLLIDER_NAME);
+                    colliderGo.layer = PhysicsLayers.onPointerEventLayer;
+                    MeshCollider collider = colliderGo.AddComponent<MeshCollider>();
+                    collider.sharedMesh = renderer.GetComponent<MeshFilter>().sharedMesh;
+                    colliderGo.transform.SetParent(renderer.transform);
+                    colliderGo.transform.ResetLocalTRS();
+                    pointerColliders.Add(collider);
+                }
```
Comment (Kinerius) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerHandler.cs:123]: Also here:

```C#
///
/// Rich comments here
///
CreatePointerEventColliders() 
```

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerHandler.cs [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+                for (int i = 0; i < renderers.Count; i++)
+                {
+                    renderersComponent.AddRenderer(scene, entity, renderers[i]);
+                }
+                // TODO: modify Animator component to remove `AddShapeReady` usage
+                dataStoreEcs7.AddShapeReady(entity.entityId, gameObject);
+                dataStoreEcs7.RemovePendingResource(scene.sceneData.id, prevLoadedGltf);
+            };
+
+            gltfLoader.OnFailEvent += exception =>
```
Comment (Kinerius) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerHandler.cs:143]: Extract method :D

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/Tests/GltfContainerHandlerShould.cs [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+            model = new PBGltfContainer()
+            {
+                Src = "sharknado"
+            };
+
+            handler.OnComponentModelUpdated(scene, entity, model);
+            yield return new WaitUntil(() => handler.gltfLoader.isFinished);
+
+            Assert.AreEqual("shark_skeleton", handler.gameObject.transform.GetChild(0).GetChild(0).GetChild(0).name);
```
Comment (Kinerius) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/Tests/GltfContainerHandlerShould.cs:90]: I would add a comment here for future devs to understand why we are navigating through child 3 times, what if the hierarchy changes? ( It's going to change with GLTFast importer )

**Review by Kinerius — CHANGES_REQUESTED** [review]
Looking good, I added some clean code refactor suggestions

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/Tests/GltfContainerHandlerShould.cs (hunk 5 above)
Comment (pbosio) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/Tests/GltfContainerHandlerShould.cs:92]: If GLTF hierarchy changes then the test must be changed accordingly

**Review by Kinerius — APPROVED** [review]

**Pull request merged**